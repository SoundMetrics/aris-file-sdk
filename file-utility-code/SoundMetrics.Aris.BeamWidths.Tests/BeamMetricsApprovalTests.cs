using ApprovalTests;
using ApprovalTests.Namers;
using ApprovalTests.Reporters;
using Microsoft.VisualStudio.TestTools.UnitTesting;
using SoundMetrics.Aris.Headers;
using System.Text;

namespace SoundMetrics.Aris.BeamWidths.Tests
{
    [TestClass]
    [UseApprovalSubdirectory("approval-files"), UseReporter(typeof(DiffReporter))]
    public sealed class BeamMetricsApprovalTests
    {
        [TestMethod]
        public void BeamWidths_ARIS3000_128()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 128, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS3000_64()
            => RunTest(new BeamConfig(ArisSystemType.Aris3000, 64, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 48, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1800_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.None));

        [TestMethod]
        public void BeamWidths_ARIS1200_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_48()
            => RunTest(new BeamConfig(ArisSystemType.Aris1200, 48, LensType.Telephoto));

        [TestMethod]
        public void BeamWidths_ARIS1800_Telephoto_96()
            => RunTest(new BeamConfig(ArisSystemType.Aris1800, 96, LensType.Telephoto));

        private static void RunTest(BeamConfig cfg)
        {
            var stringOutput = BuildStringOutput(cfg);
            Approvals.Verify(stringOutput);
        }

        private static string BuildStringOutput(BeamConfig cfg)
        {
            var beamInfos = Metrics.GetBeamInformation(cfg);
            var stringOutput = BuildOutput(beamInfos);
            return stringOutput;

            static string BuildOutput(BeamInfo[] beamInfos)
            {
                var buffer = new StringBuilder();
                buffer.AppendLine($"# of BeamInfo members: [{beamInfos.Length}]");

                foreach (var beamInfo in beamInfos)
                {
                    buffer.AppendLine(beamInfo.ToString());
                }

                return buffer.ToString();
            }
        }
    }
}
