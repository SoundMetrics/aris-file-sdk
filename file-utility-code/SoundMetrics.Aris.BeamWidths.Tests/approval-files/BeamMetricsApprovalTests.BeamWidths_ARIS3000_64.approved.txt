# of BeamInfo members: [64]
BeamInfo { BeamNumber = 0, Center = -14.8872, Left = -15.1175, Right = -14.6569 }
BeamInfo { BeamNumber = 1, Center = -14.4266, Left = -14.6569, Right = -14.1952 }
BeamInfo { BeamNumber = 2, Center = -13.9637, Left = -14.1952, Right = -13.7314 }
BeamInfo { BeamNumber = 3, Center = -13.499, Left = -13.7314, Right = -13.2661 }
BeamInfo { BeamNumber = 4, Center = -13.0332, Left = -13.2661, Right = -12.7999 }
BeamInfo { BeamNumber = 5, Center = -12.5665, Left = -12.7999, Right = -12.3326 }
BeamInfo { BeamNumber = 6, Center = -12.0986, Left = -12.3326, Right = -11.8641 }
BeamInfo { BeamNumber = 7, Center = -11.6295, Left = -11.8641, Right = -11.3944 }
BeamInfo { BeamNumber = 8, Center = -11.1593, Left = -11.3944, Right = -10.9238 }
BeamInfo { BeamNumber = 9, Center = -10.6883, Left = -10.9238, Right = -10.4524 }
BeamInfo { BeamNumber = 10, Center = -10.2165, Left = -10.4524, Right = -9.9802 }
BeamInfo { BeamNumber = 11, Center = -9.7439, Left = -9.9802, Right = -9.5073 }
BeamInfo { BeamNumber = 12, Center = -9.2706, Left = -9.5073, Right = -9.0338 }
BeamInfo { BeamNumber = 13, Center = -8.7969, Left = -9.0338, Right = -8.5599 }
BeamInfo { BeamNumber = 14, Center = -8.3228, Left = -8.5599, Right = -8.0857 }
BeamInfo { BeamNumber = 15, Center = -7.8486, Left = -8.0857, Right = -7.6114 }
BeamInfo { BeamNumber = 16, Center = -7.3742, Left = -7.6114, Right = -7.137 }
BeamInfo { BeamNumber = 17, Center = -6.8997, Left = -7.137, Right = -6.6624 }
BeamInfo { BeamNumber = 18, Center = -6.425, Left = -6.6624, Right = -6.1876 }
BeamInfo { BeamNumber = 19, Center = -5.9502, Left = -6.1876, Right = -5.7128 }
BeamInfo { BeamNumber = 20, Center = -5.4753, Left = -5.7128, Right = -5.2378 }
BeamInfo { BeamNumber = 21, Center = -5.0003, Left = -5.2378, Right = -4.7627 }
BeamInfo { BeamNumber = 22, Center = -4.5251, Left = -4.7627, Right = -4.2875 }
BeamInfo { BeamNumber = 23, Center = -4.0499, Left = -4.2875, Right = -3.8122 }
BeamInfo { BeamNumber = 24, Center = -3.5745, Left = -3.8122, Right = -3.3368 }
BeamInfo { BeamNumber = 25, Center = -3.0991, Left = -3.3368, Right = -2.8613 }
BeamInfo { BeamNumber = 26, Center = -2.6234, Left = -2.8613, Right = -2.3854 }
BeamInfo { BeamNumber = 27, Center = -2.1473, Left = -2.3854, Right = -1.9091 }
BeamInfo { BeamNumber = 28, Center = -1.6708, Left = -1.9091, Right = -1.4324 }
BeamInfo { BeamNumber = 29, Center = -1.1939, Left = -1.4324, Right = -0.9553 }
BeamInfo { BeamNumber = 30, Center = -0.7166, Left = -0.9553, Right = -0.4778 }
BeamInfo { BeamNumber = 31, Center = -0.2389, Left = -0.4778, Right = 0 }
BeamInfo { BeamNumber = 32, Center = 0.2389, Left = 0, Right = 0.4778 }
BeamInfo { BeamNumber = 33, Center = 0.7166, Left = 0.4778, Right = 0.9553 }
BeamInfo { BeamNumber = 34, Center = 1.1939, Left = 0.9553, Right = 1.4324 }
BeamInfo { BeamNumber = 35, Center = 1.6708, Left = 1.4324, Right = 1.9091 }
BeamInfo { BeamNumber = 36, Center = 2.1473, Left = 1.9091, Right = 2.3854 }
BeamInfo { BeamNumber = 37, Center = 2.6234, Left = 2.3854, Right = 2.8613 }
BeamInfo { BeamNumber = 38, Center = 3.0991, Left = 2.8613, Right = 3.3368 }
BeamInfo { BeamNumber = 39, Center = 3.5745, Left = 3.3368, Right = 3.8122 }
BeamInfo { BeamNumber = 40, Center = 4.0499, Left = 3.8122, Right = 4.2875 }
BeamInfo { BeamNumber = 41, Center = 4.5251, Left = 4.2875, Right = 4.7627 }
BeamInfo { BeamNumber = 42, Center = 5.0003, Left = 4.7627, Right = 5.2378 }
BeamInfo { BeamNumber = 43, Center = 5.4753, Left = 5.2378, Right = 5.7128 }
BeamInfo { BeamNumber = 44, Center = 5.9502, Left = 5.7128, Right = 6.1876 }
BeamInfo { BeamNumber = 45, Center = 6.425, Left = 6.1876, Right = 6.6624 }
BeamInfo { BeamNumber = 46, Center = 6.8997, Left = 6.6624, Right = 7.137 }
BeamInfo { BeamNumber = 47, Center = 7.3742, Left = 7.137, Right = 7.6114 }
BeamInfo { BeamNumber = 48, Center = 7.8486, Left = 7.6114, Right = 8.0857 }
BeamInfo { BeamNumber = 49, Center = 8.3228, Left = 8.0857, Right = 8.5599 }
BeamInfo { BeamNumber = 50, Center = 8.7969, Left = 8.5599, Right = 9.0338 }
BeamInfo { BeamNumber = 51, Center = 9.2706, Left = 9.0338, Right = 9.5073 }
BeamInfo { BeamNumber = 52, Center = 9.7439, Left = 9.5073, Right = 9.9802 }
BeamInfo { BeamNumber = 53, Center = 10.2165, Left = 9.9802, Right = 10.4524 }
BeamInfo { BeamNumber = 54, Center = 10.6883, Left = 10.4524, Right = 10.9238 }
BeamInfo { BeamNumber = 55, Center = 11.1593, Left = 10.9238, Right = 11.3944 }
BeamInfo { BeamNumber = 56, Center = 11.6295, Left = 11.3944, Right = 11.8641 }
BeamInfo { BeamNumber = 57, Center = 12.0986, Left = 11.8641, Right = 12.3326 }
BeamInfo { BeamNumber = 58, Center = 12.5665, Left = 12.3326, Right = 12.7999 }
BeamInfo { BeamNumber = 59, Center = 13.0332, Left = 12.7999, Right = 13.2661 }
BeamInfo { BeamNumber = 60, Center = 13.499, Left = 13.2661, Right = 13.7314 }
BeamInfo { BeamNumber = 61, Center = 13.9637, Left = 13.7314, Right = 14.1952 }
BeamInfo { BeamNumber = 62, Center = 14.4266, Left = 14.1952, Right = 14.6569 }
BeamInfo { BeamNumber = 63, Center = 14.8872, Left = 14.6569, Right = 15.1175 }
